tools/clang/test/CodeGenSPIRV/method.byte-address-buffer.templated-load.scalar.hlsl

```diff
@@ -6,22 +6,27 @@ ByteAddressBuffer buf;
                                  : SV_DispatchThreadId) {
   // ********* 16-bit scalar ********************
 
+  // CHECK:   [[byte:%\d+]] = OpUMod %uint {{%\d+}} %uint_4
+  // CHECK:   [[bits:%\d+]] = OpIMul %uint [[byte]] %uint_8
   // CHECK:    [[ptr:%\d+]] = OpAccessChain %_ptr_Uniform_uint %buf %uint_0 {{%\d+}}
   // CHECK:   [[uint:%\d+]] = OpLoad %uint [[ptr]]
-  // CHECK: [[ushort:%\d+]] = OpUConvert %ushort [[uint]]
+  // CHECK:  [[shift:%\d+]] = OpShiftRightLogical %uint [[uint]] [[bits]]
+  // CHECK: [[ushort:%\d+]] = OpUConvert %ushort [[shift]]
   // CHECK:                   OpStore %u16 [[ushort]]
   uint16_t u16 = buf.Load<uint16_t>(tid.x);
```

Comment:
Can you add a test similar to the one in the bug, explicitly indexing into the buffer at byte offset 0 and 2? It would be helpful to have a test where we can CHECK that the actual numeric values in the shift and mask operations are what we expect. It's harder to see with these ones what was incorrect before and correct now.

Reply:
Sure, I've added it to the top of `method.byte-address-buffer.templated-load.scalar.hlsl`